transport/x/grpc/config.go

```diff
@@ -127,6 +131,18 @@ func newTransportSpec(opts ...Option) (*transportSpec, error) {
 
 func (t *transportSpec) buildTransport(transportConfig *TransportConfig, _ *yarpcconfig.Kit) (transport.Transport, error) {
 	transportOptions := newTransportOptions(t.TransportOptions)
+	if transportConfig.ServerMaxRecvMsgSize > 0 {
+		transportOptions.serverMaxRecvMsgSize = transportConfig.ServerMaxRecvMsgSize
+	}
+	if transportConfig.ServerMaxSendMsgSize > 0 {
+		transportOptions.serverMaxSendMsgSize = transportConfig.ServerMaxSendMsgSize
```

Comment:
readability nit: you can declare variables in if statements to help
readability here.

```go
if size := transportConfig.ServerMaxRecvMsgSize; size > 0 {
  transportOptions.serverMaxRecvMsgSize = size
}
```

Reply:
Also, changing private fields of `transportOptions` is fragile, especially
since it already provides a means to change these fields via options. I
suggest using the options,

```go
opts := t.TransportOptions
if s := cfg.ServerMaxRecvMsgSize; s > 0 {
  opts = append(opts, ServerMaxRecvMsgSize(s))
}
```

Finally, if you applied defaults in the Option implementation or in
newTransportOptions after applying the Options, you could forego these checks
and indiscriminately add the size options.

Reply:
I buy most of this - on the creating the variable, I actually like it the existing way more, I was CS-raised not to declare extra state if not necessary and it just feels dirty. I buy the options appending.